### PR TITLE
pop up

### PR DESCRIPTION
**Pull request by kipiiler**
Author: Hoang Nguyen

## What changes were made?
Made a confirmation pop-up before pairing & listing signup pool status

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -167,3 +167,8 @@ export interface RawTechnicalQuestion {
   follow_ups: string;
   source: string;
 }
+
+export interface InterviewStatus {
```
Comment (elimelt) [inline src/types.ts:171]: InterviewSignupStatus or InterviewPoolStatus instead

**Review by elimelt — COMMENTED** [review]
LGTM besides comment, might get confusing

File: src/components/debug/PairingDashboard.tsx [hunk 2]
```diff
@@ -9,37 +9,70 @@ import {
...
   const toast = useToast();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
+  useEffect(() => {
+    api
+      .get("/interview/status/")
+      .then((res) => {
+        setLoading(true);
+        setSignal(!signal);
+        console.log(res.data);
```
Comment (ShawnCollinge) [inline src/components/debug/PairingDashboard.tsx:42]: Let's not have console.logs in prod please

File: src/components/debug/PairingDashboard.tsx [hunk 3]
```diff
@@ -9,37 +9,70 @@ import {
...
+        if (res.data) {
+          setSignUpData(res.data);
+        }
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (ShawnCollinge) [inline src/components/debug/PairingDashboard.tsx:48]: No console.log please.

File: src/components/debug/PairingDashboard.tsx [hunk 4]
```diff
@@ -9,37 +9,70 @@ import {
...
 
 const PairingDashboard = () => {
   const [loading, setLoading] = useState(false);
   const [signal, setSignal] = useState(false);
-  const [response, setResponse] = useState('');
+  const [response, setResponse] = useState("");
+
+  const [signupData, setSignUpData] = useState<InterViewPoolStatus>({
+    members: [],
+    number_sign_up: 0,
```
Comment (ShawnCollinge) [inline src/components/debug/PairingDashboard.tsx:31]: camelCase

**Review by ShawnCollinge — CHANGES_REQUESTED** [review]

File: src/components/debug/PairingDashboard.tsx [hunk 5]
```diff
@@ -19,33 +19,34 @@ import {
...
+import { devPrint } from "../utils/RandomUtils";
 
 const PairingDashboard = () => {
   const [loading, setLoading] = useState(false);
   const [signal, setSignal] = useState(false);
   const [response, setResponse] = useState("");
 
-  const [signupData, setSignUpData] = useState<InterViewPoolStatus>({
+  const [signupData, setSignUpData] = useState<InterviewPoolStatus>({
+    numberSignUp: 0,
```
Comment (ShawnCollinge) [inline src/components/debug/PairingDashboard.tsx:32]: You don't need to set this to 0 as you're already checking below if it's undefined and returning 0 if it is.

File: src/services/api.ts [hunk 6]
```diff
@@ -1,31 +1,31 @@
...
   }
 };
 
 api.interceptors.request.use(async (config) => {
-  if (config.url === '/auth/csrf/') {
+  if (config.url === "/auth/csrf/") {
     return config;
   }
-  if (!api.defaults.headers.common['X-CSRFToken']) {
+  if (!api.defaults.headers.common["X-CSRFToken"]) {
```
Comment (ShawnCollinge) [inline src/services/api.ts:28]: Please rerun the correct linter on these files.

File: src/services/interview.ts [hunk 7]
```diff
@@ -6,13 +6,24 @@ import {
...
-} from './question';
+} from "./question";
+
+function serializeInterviewPoolStatus(
+  data: RawInterViewPoolStatus
+): InterviewPoolStatus {
+  return {
+    numberSignUp: data.number_sign_up,
+    members: data.members,
+  } as InterviewPoolStatus;
```
Comment (ShawnCollinge) [inline src/services/interview.ts:25]: Why are we casting here?

File: src/services/interview.ts [hunk 8]
```diff
@@ -6,13 +6,24 @@ import {
...
+import api from "./api";
+import { deserializeMember } from "./member";
 import {
   deserializeTechnicalQuestion,
   deserializeBehavioralQuestion,
-} from './question';
+} from "./question";
+
+function serializeInterviewPoolStatus(
+  data: RawInterViewPoolStatus
```
Comment (ShawnCollinge) [inline src/services/interview.ts:20]: Let's use the same format we've used elsewhere for serializing and deserializing.

**Review by elimelt — APPROVED** [review]
LGTM

**Pull request merged**